src/zimwriterfs/zimwriterfs.cpp

```diff
@@ -346,6 +349,26 @@ void create_zim()
 {
   ZimCreatorFS zimCreator(directoryPath, welcome, isVerbose(), uniqueNamespace, zstdFlag);
 
+  if (zimPath.size() >= (MAXPATHLEN-1))
+    throw std::invalid_argument("Target .zim file path is too long");
+
+  // dirname() can modify the 'buf', no c_str() here.
+  char buf[MAXPATHLEN];
+  strncpy(buf, zimPath.c_str(), sizeof(buf));
```

Comment:
We are reading after zimPath here. (sizeof(buf) > zimPath.size()).
This is not a problem as c_str() ensure there is a `\0` at the end of the string, but I dislike it anyway :)

We don't need to initialize `buf` with zimPath. We can initialize `zimdir` with `zimPath.c_str()`.

Reply:
Not exactly. zimdir is initialized with _the result_ of dirname(), but we cannot pass c_str() to dirname() because it takes `char *` (not `const char *`).